:bookmark: release 3.5.0-alpha.0 (#5945)
open-formulieren/open-forms: CHANGELOG.rst

    The Dutch version of this changelog can be found :ref:`here <changelog-nl>`.

> 3.5.0-alpha.0 (2026-02-05)

Author: Reminder: update date before release

> ==========================

Reviewer: ```suggestion
3.5.0-alpha.0 (2026-02-11)
==========================
```

        # in the container via ``docker exec`` or ``kubectl exec``:
        python /app/bin/check_disable_next_logic_action.py

New features
------------

> * JCC integration:

Reviewer: ```suggestion
* JCC REST API integration:
```

>   - [:backend:`5690`] Added the extra field description for a product in appointments.

Reviewer: ```suggestion
  - [:backend:`5690`] The additional product information (HTML) is now exposed to the SDK.
```

>   - [:backend:`5696`] Implemented JCC API calls

Reviewer: ```suggestion
  - [:backend:`5696`] Implemented the remaining API calls for the appointment flow.
```

* Performance:

  - [:backend:`5861`] Added step selection field to the "disable next" action, which indicates on which step
    the corresponding rule should be executed.
  - [:backend:`2409`] Added support for variable extraction from if statements and for loops
    in Django templates and refactored form logic serializer.

* [:backend:`5287`] The health check mechanisms are added for all containers:

  - ``/_healthz/*`` endpoints are added for the django application

>   - liveness files for celery and celery beat

Reviewer: ```suggestion
```
details are available in the docs already.

* [:backend:`5450`] The support for Open telemetry tracing is extended. Now all redis and PostgreSQL queries,
  celery tasks and external requests are auto instrumented, we also included custom traces for

>   interesting parts of Open Forms.

Reviewer: ```suggestion
* [:backend:`5450`] Distributed tracing is now mostly stable. The following component traces are now also
  available:
  
  - Postgres queries
  - Redis queries
  - Celery tasks
  - Outgoing requests to external services
  - Application-specific traces
```

* [:backend:`5319`] Login option for organizations with OpenID Connect can be now dinamically

>   displayed depending on the ``auth_visible`` query param.

Author: I've renamed it to snake_case in this [commit](https://github.com/open-formulieren/open-forms/pull/5907/commits/6fb3bee512d20729e5f57e23f572c92319b1b27d) to make it in style with our other query params

* [:backend:`5820`] We now support retrieving custom static translation assets from the backend.
  The new enpoint is added to upload the custom translations.

Bugfixes
--------

> * [:backend:`5902`] Refactored hadling of prefill data and moved it from submission to state.

Reviewer: not a bugfix - should be moved under the performance section in the features part.

```suggestion
* [:backend:`5902`] Cleaned up how we handle prefilled data.
```

> * [:backend:`5893`] Fixed ``amount`` attribute of appointment Product not being set.

Reviewer: ```suggestion
* [:backend:`5893`] Fixed ``amount`` attribute of appointment product not being set when processing query parameters.
```

Project maintenance
-------------------

* [:backend:`5879`] Fixed various incorrect configurations for component definitions.
* Added a test for prefill validation of nested data to make sure that the data is properly validated.
* Cleaned up inconsistent test setups.
* Updated `pyright` and improved type checking.
* Replaced bleach with nh3 sanitizer.
* Updated dependencies to their latest security releases:

  - protobuf
  - weasyprint

>   - copy-webpack-plugin

Reviewer: 3.4.1 is based on `stable/3.4.x`, not on `main`. The previous release (tag) from this branch is `3.4.0`

>   - copy-webpack-plugin

Author: Please ignore it, I missed your comment that we include 3.4.1 changes here

> * Upgraded to storybook 9.1.17.

Reviewer: ```suggestion
* Upgraded to storybook 9.
```

    The automatic migrations update can reliably determine the step most of the time. For the forms
    to manually review, see the upgrade procedure notes.
  - [:backend:`2409`] Open Forms can now statically determine which input and output variables are used
    in logic rules.

* [:backend:`5287`] Low-level health check mechanisms are now exposed in all Open Forms components.

* [:backend:`5450`] Distributed tracing is now mostly stable. The following component traces are now also
  available:

  - Postgres queries
  - Redis queries
  - Celery tasks
  - Outgoing requests to external services
  - Application-specific traces
* [:backend:`5319`] The "Organization login via OpenID Connect" can now be hidden by default. Use the ``authVisible=all`` query parameter to make it visible in the SDK.
* [:backend:`5820`] Added endpoint to retrieve the overridden frontend translations.
* [:backend:`5553`] The organization branch number ('vestigingsnummer') is added for login, prefill and
  registration.

Bugfixes
--------

* [:backend:`5902`] Cleaned up how we handle prefilled data.

> * [:backend:`5893`] Fixed ``amount`` attribute of appointment product not being set when processing query parameters.

Author: Oh I missed your comment about it, I didn't know we include them in 3.5 version. I'll add them!
